components/catalogs/back/src/Application/Service/ProductMapperInterface.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\Catalogs\Application\Service;
+
+use Akeneo\Catalogs\Application\Persistence\Catalog\Product\GetRawProductsQueryInterface;
+use Akeneo\Catalogs\Domain\Catalog;
+use Akeneo\Catalogs\ServiceAPI\Query\GetMappedProductsQuery;
+
+/**
+ * @copyright 2022 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ *
+ * @phpstan-import-type MappedProduct from GetMappedProductsQuery
+ * @phpstan-import-type RawProduct from GetRawProductsQueryInterface
+ */
+interface ProductMapperInterface
+{
+    /**
+     * @param array<RawProduct> $products
+     *
+     * @return array<MappedProduct>
+     */
+    public function getMappedProducts(array $products, Catalog $catalog): array;
+
+    /**
+     * @param RawProduct $product
+     *
+     * @return MappedProduct
+     */
+    public function getMappedProduct(array $product, Catalog $catalog): array;
```

Comment:
```suggestion
    /**
     * @param RawProduct $product
     *
     * @return MappedProduct
     */
    public function getMappedProduct(array $product, Catalog $catalog): array;
```
I would rather have only one method and do the loop outside.